subprojects/logging/src/main/java/org/gradle/internal/logging/progress/DefaultProgressLoggerFactory.java

```diff
@@ -40,25 +42,40 @@ public ProgressLogger newOperation(Class loggerCategory) {
         return newOperation(loggerCategory.getName());
     }
 
+    public ProgressLogger newOperation(Class loggerCategory, OperationIdentifier operationIdentifier) {
+        return newOperation(loggerCategory.getName());
+    }
+
     public ProgressLogger newOperation(String loggerCategory) {
-        return init(loggerCategory, null);
+        return init(loggerCategory, null, null);
+    }
+
+    public ProgressLogger newOperation(String loggerCategory, OperationIdentifier operationIdentifier) {
+        return init(loggerCategory, null, operationIdentifier);
     }
 
     public ProgressLogger newOperation(Class loggerClass, ProgressLogger parent) {
-        return init(loggerClass.toString(), parent);
+        return init(loggerClass.toString(), parent, null);
     }
 
-    private ProgressLogger init(String loggerCategory, ProgressLogger parentOperation) {
+    private ProgressLogger init(String loggerCategory, @Nullable ProgressLogger parentOperation, @Nullable OperationIdentifier operationIdentifier) {
         if (parentOperation != null && !(parentOperation instanceof ProgressLoggerImpl)) {
             throw new IllegalArgumentException("Unexpected parent logger.");
         }
-        return new ProgressLoggerImpl((ProgressLoggerImpl) parentOperation, nextId.getAndIncrement(), loggerCategory, progressListener, timeProvider);
+
+        // Assign fake operation ID so that we can still complete logging headers correctly.
+        // This will go away as we force all progress loggers to provide a build operation id
+        if (operationIdentifier == null) {
+            // Decrement from -1 to avoid any conflict between this ID and build operation IDs
```

Comment:
I never want a "build operation" OperationId to _ever_ be equal to a progress logger Operation id, because I'd hate for log grouping to be messed up because somehow a fake operation id for progress logging had the same ID as a real build operation. Make sense?